services:

  nginx-pm:
    #FIXME latest can b dangerous,have broken stuff b4, should be a set build
    image: 'jc21/nginx-proxy-manager:latest'
    restart: unless-stopped
    ports:
      - '80:80'
      - '443:443'
      - ${PORT_NGINX_PM}:81 #web console
    volumes:
      - ./data/shared/letsencrypt:/etc/letsencrypt
      - ./data/shared/nginx-pm:/data
      # add custom config in /data/nginx/custom
      # https://nginxproxymanager.com/advanced-config/#custom-nginx-configurations
    healthcheck:
      test: [ "CMD-SHELL", "curl -f http://localhost:80 || exit 1" ]
      interval: 30s
      timeout: 10s
      retries: 5
      start_period: 30s


  portainer: #inner port :9000
    image: portainer/portainer-ce:latest
    container_name: portainer
    restart: always
    security_opt:
      - no-new-privileges:true
    volumes:
      - ./portainer-readonly/run/docker.sock:/var/run/docker.sock:ro
      - ../shared/portainer:/data


  mysql8: #inner port :3306
    image: mysql:8.0
    environment:
      - MYSQL_ROOT_PASSWORD=${DB_ROOT_PASSWORD}
    volumes:
      - ../shared/mysql:/var/lib/mysql
      - ./mysql.cnf:/etc/mysql/conf.d/my.cnf
    restart: unless-stopped
    healthcheck:
      test: [ "CMD", "mysqladmin", "ping", "-h", "localhost" ]
      interval: 30s
      timeout: 10s
      retries: 5
      start_period: 30s


  phpmyadmin: #inner port :80
    image: phpmyadmin
    restart: always
    environment:
      PMA_HOST: ${DB_HOST}
      PMA_PORT: ${DB_PORT}
      PMA_ARBITRARY: 0
    healthcheck:
      test: [ "CMD-SHELL", "curl -f http://localhost:80 || exit 1" ]
      interval: 30s
      timeout: 10s
      retries: 5
      start_period: 30s
